enterprise/api_client/enterprise.py

```diff
@@ -0,0 +1,50 @@
+
+# -*- coding: utf-8 -*-
+"""
+Client for communicating with the Enterprise API.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from django.conf import settings
+from django.core.cache import cache
+
+from enterprise.api_client.lms import LmsApiClient
+from enterprise.utils import get_cache_key, traverse_pagination
+
+
+class EnterpriseApiClient(LmsApiClient):
+    """
+    Object builds an API client to make calls to the Enterprise API.
+    """
+
+    API_BASE_URL = settings.LMS_ROOT_URL + 'enterprise/api/v1/'
+
+    def get_all_enterprise_courses(self, enterprise_customer):
+        """
+        Query the Enter API for the course details of the given course_id.
+
+        Arguments:
+            enterprise_customer (Enterprise Customer): Enterprise customer for fetching courses
+
+        Returns:
+            dict: A dictionary containing details about the course, in an enrollment context (allowed modes, etc.)
+        """
+        api_resource_name = 'enterprise-customer'
```

Comment:
I would recommend taking a closer look at this logic because I just tried to run the job on the sandbox and got this error:

```
edxapp@business:~/edx-platform$ ./manage.py lms transmit_courseware_data --catalog_user staff --settings=aws
2017-08-18 11:11:26,268 INFO 28320 [dd.dogapi] dog_stats_api.py:66 - Initializing dog api to use statsd: localhost, 8125
2017-08-18 11:11:26,614 INFO 28320 [integrated_channels.integrated_channel.management.commands.transmit_courseware_data] transmit_courseware_data.py:77 - Processing courses for integrated channel using configuration: <SAPSuccessFactorsEnterpriseCustomerConfiguration for Enterprise SuccessFactors Company>
Traceback (most recent call last):
  File "./manage.py", line 116, in <module>
    execute_from_command_line([sys.argv[0]] + django_args)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 346, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/base.py", line 394, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/management/base.py", line 445, in execute
    output = self.handle(*args, **options)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/management/commands/transmit_courseware_data.py", line 60, in handle
    send_data_task.delay(username, channel_code, channel_pk)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/management/commands/__init__.py", line 24, in no_delay
    func(*args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/management/commands/transmit_courseware_data.py", line 80, in send_data_task
    channel.transmit_course_data(user)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/models.py", line 120, in transmit_course_data
    course_data_exporter = self.get_course_data_exporter(user)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/sap_success_factors/models.py", line 144, in get_course_data_exporter
    return SapCourseExporter(user, self)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/sap_success_factors/utils.py", line 154, in __init__
    super(SapCourseExporter, self).__init__(user, plugin_configuration)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/course_metadata.py", line 78, in __init__
    for course_run in get_course_runs(self.enterprise_customer):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/integrated_channels/integrated_channel/course_metadata.py", line 51, in get_course_runs
    enterprise_courses = client.get_all_enterprise_courses(enterprise_customer)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/enterprise/api_client/enterprise.py", line 39, in get_all_enterprise_courses
    endpoint = getattr(self, api_resource_name)
AttributeError: 'EnterpriseApiClient' object has no attribute 'enterprise-customer'
```